src/core/workload.py

```diff
@@ -8,17 +8,24 @@
 import string
 from abc import ABC, abstractmethod
 
-from literals import PATHS
+from literals import PATHS, Role
 
 
-class KafkaPaths:
+class CharmedKafkaPaths:
     """Object to store common paths for Kafka."""
 
-    def __init__(self):
-        self.conf_path = PATHS["CONF"]
-        self.data_path = PATHS["DATA"]
-        self.binaries_path = PATHS["BIN"]
-        self.logs_path = PATHS["LOGS"]
+    def __init__(self, role: Role):
+
+        match role:
+            case Role.BROKER:
+                sub_path = "kafka"
+            case Role.PARTITIONER:
+                sub_path = "cruise-control"
```

Comment:
**todo:** I think these need to be hard-coded somewhere in literals. I think there's value in having the full path prefix as a configuration global accessible everywhere.

Reply:
Something like:
```python
PATHS = {
    "kafka": {
        "CONF": f"/var/snap/{SNAP_NAME}/current/etc/kafka",
        "LOGS": f"/var/snap/{SNAP_NAME}/common/var/log/kafka",
        "DATA": f"/var/snap/{SNAP_NAME}/common/var/lib/kafka",
        "BIN": f"/snap/{SNAP_NAME}/current/opt/kafka",
    },
    "cruise-control": {
        "CONF": f"/var/snap/{SNAP_NAME}/current/etc/cruise-control",
        "LOGS": f"/var/snap/{SNAP_NAME}/common/var/log/cruise-control",
        "DATA": f"/var/snap/{SNAP_NAME}/common/var/lib/cruise-control",
        "BIN": f"/snap/{SNAP_NAME}/current/opt/cruise-control",
    },
}

Reply:
Done in 07acec4d0df854b14313445566757c0f82555183. Thanks for your input. I definitely got lost in some unnecessary complexity.